calculator.go

```diff
@@ -1,13 +1,89 @@
 // Package calculator provides a library for simple calculations in Go.
 package calculator
 
+import (
+	"errors"
+	"fmt"
+	"math"
+)
+
 // Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+func Add(nums ...float64) float64 {
+	var result float64 = nums[0]
+
+	for _, n := range nums[1:] {
+		result = result + n
+	}
+	return result
 }
 
 // Subtract takes two numbers and returns the result of subtracting the second
 // from the first.
-func Subtract(a, b float64) float64 {
-	return b - a
+func Subtract(nums ...float64) float64 {
+	var result float64 = nums[0]
+
+	for _, n := range nums[1:] {
+		result = result - n
+	}
+	return result
+}
+
+// Multiply takes two numbers and returns the result of multiplying them
+func Multiply(nums ...float64) float64 {
+	var result float64 = 1
+
+	for _, n := range nums {
+		result = result * n
+	}
+
+	return result
+}
+
+// Divide takes two numbers and returns the result of dividing the first by the second
+func Divide(nums ...float64) (float64, error) {
+	var result float64 = nums[0]
+	var err error
+
+	for _, n := range nums[1:] {
+		if n == 0 {
+			err = errors.New("Error: division by 0 is not allowed")
```

Comment:
We could (and should) return right away here, since we know there's no point doing any more calculations. The convention in the error case is to return zero for the `float64` value.

Reply:
Addressed in 1f30bd8